### ISSUE TITLE
Tree: should have more screen reader friendly structure

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#92564]
From @leonardder 

So, one entry looks as follows:

```
			<div class="monaco-list-row focused" role="treeitem" data-index="0" data-last-element="false"
				aria-setsize="2" aria-posinset="1" id="list_id_3_0" aria-selected="false" aria-label="Child1"
				aria-level="1" aria-expanded="true" draggable="true" style="top: 0px; height: 22px; line-height: 22px;">
				...
			</div>
```

Would it be possible to add an additional div within every monaco-list-row or monaco-tl-contents div with a role of group that has aria-owns containing a space separated list with IDs of direct descendants? This way, we might be able to convince the accessibility tree that there's a hierarchie.

If you don't understand what I mean, I can work out an example in HTML to explain this.
More details here https://github.com/microsoft/vscode/issues/91061

**Comment by isidorn** [comment]
I think this would be quite some undertaking on the tree level to make it more screen reader friendly as you suggest @leonardder 
@joaomoreno would know best and currently he is on a paternity leave. Thus I suggest to wait for him to get his feedback.

**Comment by leonardder** [comment]
May be we should prototype something first?

**Comment by isidorn** [comment]
That is always an option, however I would not have the time this milestone.
And our tree is quite optimized for performance and it is very complex. Just changing this might prove to be some demanding work. In the end I am not sure if it is even possible, thus waiting for @joaomoreno feedback is imho the best option atm.

**Comment by leonardder** [comment]
Actually... I made a dump of a tree view [Here](https://github.com/microsoft/vscode/files/4324364/currentTreeView.zip)

When opening that in Chrome 80.0.3987.132. oit looks like NVDA is aable to find out the position info in the tree based on the aria properties.
May be this will be fixed when switching to a newer version of Electron.

**Comment by isidorn** [comment]
@leonardder we update to electron regularly and we can keep our eyes open on this one.

**Comment by joaomoreno** [comment]
> Would it be possible to add an additional div within every monaco-list-row or monaco-tl-contents div with a role of group that has aria-owns containing a space separated list with IDs of direct descendants? This way, we might be able to convince the accessibility tree that there's a hierarchie.

Definitely not going to sacrifice performance for this.

What is the actual problem? I thought the `aria` attributes are exactly for this situation: to provide structure where there normally is none.

**Comment by leonardder** [comment]
It actually looks like this is more of a bug in Chromium that might have 
been fixed in a later version.

**Comment by joaomoreno** [comment]
Fantastic.
